sdk/storage/storage-blob-changefeed/src/BlobChangeFeedClient.ts

```diff
@@ -31,9 +44,62 @@ export class BlobChangeFeedClient {
   private _blobServiceClient: BlobServiceClient;
   private _changeFeedFactory: ChangeFeedFactory;
 
-  public constructor(blobServiceClient: BlobServiceClient) {
-    this._blobServiceClient = blobServiceClient;
+  public static appendUserAgentPrefix(options?: StoragePipelineOptions): StoragePipelineOptions {
+    if (!options) {
+      options = {};
+    }
+    if (options.userAgentOptions === undefined) {
+      options.userAgentOptions = {};
+    }
+
+    if (options.userAgentOptions.userAgentPrefix === undefined) {
+      options.userAgentOptions.userAgentPrefix = "";
+    } else if (options.userAgentOptions.userAgentPrefix !== "") {
+      // two spaces to work around as the TelemetryPolicyFactory in blob removes the first space now.
+      options.userAgentOptions.userAgentPrefix += "  ";
+    }
+    options.userAgentOptions.userAgentPrefix += `changefeed-js/${SDK_VERSION}`;
```

Comment:
Are other language using this pattern for user agent prefix?

Reply:
Yes. We had a email thread about this and we reached this agreement.